packages/transform-module/README.md

```diff
@@ -1,5 +1,200 @@
+      ])],
+      ["import-named-exports-from-me.js", new Map([
+        ["fizz", [$h_a => (fizz = $h_a)]],
+        ["buzz", [$h_a => (buzz = $h_a)]],
+      ])],
+      ["import-named-export-and-rename.js", new Map([
+        ["color", [$h_a => (colour = $h_a)]],
+      ])],
+      ["import-and-reexport-name-from-me.js", new Map([
+        ["qux", [$h_live["qux"]]]
+      ])],
+      ["import-and-export-all.js", new Map([])]
+    ]),
+    ["import-and-export-all.js"]
+  );
+
+  let $c_quuux = null;
+  $h_live.quuux($c_quuux);
+
+  const { default: $c_default } = { default: 42 };
+
+  $h_once.default($c_default);
+
+  const quux = 'Hello, World!';
+  $h_once.quux(quux);
+
+  quuux = 'Sorry for binding late!';
+})
+```
+
+For the final live binding to `quuux`, we depend on the evaluator to put a
+Proxy on the scope chain to intercept the assignment and effect an update
+to all modules that import the value.
+
+```ts
+// This is the signature of the analyze function, after composing it
+// with Babel core.
+type Analyzer = ({string: ModuleSource}) => ModuleStaticRecord
+
+type ModuleSource = string
+
+type ModuleStaticRecord = {
+  exportAlls: ExportAlls,
+  imports: Imports,
+  liveExportMap: LiveExportMap,
+  fixedExportMap: FixedExportMap,
+  functorSource: string,
+};
+
+// ExportAlls are the relative module specifiers found in directives like:
+//   export * from 'import-and-reexport-all-from-me.js';
+// These are both on the module static record and passed to the import function
+// injected into a module functor.
+// TODO remove the import argument. It does not appear to be used
+// by module instances.
+type ExportAlls = Array<RelativeModuleSpecifier>
+
+// Imports includes a key for every relative module specifier in
+// any static import declaration, including those implied by
+// export/from clauses.
+// The import names are the names from the dependency module
+// that this module will import.
+// If this module reexports names from the dependency module
+// but doesn't capture them in its own scope, the imports map
+// has an entry for the module but the array of import names is empty.
+type Imports = Object<RelativeModuleSpecifier, Array<ImportNames>)
+
+// LiveExportMap indicates which variables in this module's scope
+// need to emit updates when they change.
+type LiveExportMap = Object<ExportName, [LocalName, SetProxyTrap]>
+
+// FixedExportMap indicates which constants in this module's scope
+// need to emit updates when they are initialized.
+// FixedExportMap is an aesthetic subtype of LiveExportMap.
+// The single box around LocalName is not meaningful.
+type FixedExportMap = Object<ExportName, [LocalName]>
+
+// TODO this example does not clarify why local and export names
+// are both present, since they are always the same.
+// In the case of a name that was remapped, as in the
+// `color as colour` example, the transformed code accounts for
+// the difference.
+type LocalName = string
+type ExportName = string
```

Comment:
I could use a clarification here too. Is there possibly an opportunity to simplify this convention since the local name and export name are always† the same.

† I’ve not figured out how to tease a different local name and export name out of a module text. I thought it was for `import { x as y }` or `import x as y`, but I don’t think either made a difference. The functorSource takes the difference between the local scope and the module namespace into account.

Reply:
cc @michaelfig

Reply:
Yes, you're correct.  You can maybe call it something like an _ImportName_.